core/services/pipeline/graph.go

```diff
@@ -13,151 +13,64 @@ import (
 	"gonum.org/v1/gonum/graph/topo"
 )
 
-// TaskDAG fulfills the graph.DirectedGraph interface, which makes it possible
-// for us to `dot.Unmarshal(...)` a DOT string directly into it.  Once unmarshalled,
-// calling `TaskDAG#TasksInDependencyOrder()` will return the unmarshaled tasks.
-// NOTE: We only permit one child
-type TaskDAG struct {
+// tree fulfills the graph.DirectedGraph interface, which makes it possible
+// for us to `dot.Unmarshal(...)` a DOT string directly into it.
+type Tree struct {
```

Comment:
Note: I wanted to completely hide the tree representation and only expose `Pipeline`/`Task`, but the job migration builder directly interacts with these to build a `dot` on the fly.